src/middlewared/middlewared/utils/osc/linux/mount.py

```diff
@@ -27,30 +28,79 @@ def __parse_mntent(line, out_dict):
         'fs_type': fstype,
         'mount_source': mnt_src.replace('\\040', ' '),
         'super_opts': super_opts.upper().split(','),
-    }})
+    }
 
 
-def getmntinfo(dev_id=None):
-    """
-    Get mount information. returns dictionary indexed by dev_t.
-    User can optionally specify dev_t for faster lookup of single
-    device.
-    """
+def __parse_to_dev(line, out_dict):
+    entry = __mntent_dict(line)
+    out_dict.update({entry['device_id']['dev_t']: entry})
+
+
+def __parse_to_mnt_id(line, out_dict):
+    entry = __mntent_dict(line)
+    out_dict.update({entry['mount_id']: entry})
+
+
+def __create_tree(info, dev_id):
+    root_id = None
+    mount_id = None
+
+    for entry in info.values():
+        if not entry.get('children'):
+            entry['children'] = []
+
+        if entry['parent_id'] == 1:
+            root_id = entry['mount_id']
+            continue
+
+        if entry['device_id']['dev_t'] == dev_id:
+            mount_id = entry['mount_id']
+
+        parent = info[entry['parent_id']]
+        if not parent.get('children'):
+            parent['children'] = [entry]
+        else:
+            parent['children'].append(entry)
+
+    return info[mount_id or root_id]
+
+
+def __iter_mountinfo(dev_id=None, callback=None, private_data=None):
     if dev_id:
         maj_min = f'{os.major(dev_id)}:{os.minor(dev_id)}'
     else:
         maj_min = None
 
-    out = {}
     with open('/proc/self/mountinfo') as f:
         for line in f:
             if maj_min:
                 if line.find(maj_min) == -1:
                     continue
 
-                __parse_mntent(line, out)
+                callback(line, private_data)
                 break
 
-            __parse_mntent(line, out)
+            callback(line, private_data)
+
+
+def getmntinfo(dev_id=None):
+    """
+    Get mount information. returns dictionary indexed by dev_t.
+    User can optionally specify dev_t for faster lookup of single
+    device.
+    """
+    out = {}
```

Comment:
Let's make this similar to how it's being done in `getmnttree`.

- remove unnecessary empty newlines
- change `out` to info